src/termdag.rs

```diff
@@ -21,14 +21,8 @@ pub enum Term {
 /// A hashconsing arena for [`Term`]s.
 #[derive(Clone, PartialEq, Eq, Debug, Default)]
 pub struct TermDag {
-    // think of nodes as a map from indices to Terms.
-    // invariant: the nodes map and the hashcons map are inverses.
-    // note that this implies:
-    // - no duplicates in nodes
-    // - every element of node is a key in hashcons
-    // - every key of hashcons is in nodes
-    pub nodes: Vec<Term>,
-    pub hashcons: HashMap<Term, TermId>,
+    /// A bidirectional map between deduplicated `Term`s and indices.
+    nodes: IndexSet<Term>,
```

Comment:
Would it be better to create getters for what you need? Then it matters less what it's stored as. Since there were no other uses of the field, it looked safe to make private; a method would also make that more clear. `pub` is fine too, though 🤷‍♀️

Reply:
Yeah, that sounds good to me.

Looking through the usages in Python, it seems like the only things I do with a termdag is [call `term_to_expr`](https://github.com/egraphs-good/egglog-python/blob/12d8d5f35f35a578a909a7605a40695d25bdd8ab/src/lib.rs#L15C13-L15C25) and then [lookup a term based on its id (`termdag.nodes[term_id]`)](https://github.com/egraphs-good/egglog-python/blob/12d8d5f35f35a578a909a7605a40695d25bdd8ab/python/egglog/egraph_state.py#L562)

Reply:
Looking up a term by `TermId` is already exposed via `TermDag::get`. So it sounds like what you need is already public?

What about the conversion in `convert_struct!` in your earlier comment? Would you need a `fn TermDag::iter(&self) -> impl Iterator<Item = &Term>`?

Reply:
Naw that's ok, I can just not expose the TermDag struct like that, and instead just expose a few methods on it.